packages/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,291 @@
+  [Variant.Dark]: css`
+    background-color: ${uiColors.gray.dark3};
+    color: ${uiColors.gray.light1};
+  `,
+
+  [Variant.Light]: css`
+    background-color: ${uiColors.gray.light3};
+    color: ${uiColors.gray.dark2};
+    border: 1px solid ${uiColors.gray.light2};
+  `,
+};
+
+const notchVariants = {
+  [Variant.Dark]: css`
+    background-color: ${uiColors.gray.dark3};
+    box-shadow: 0px 2px 4px ${transparentize(0.85, uiColors.black)};
+  `,
+  [Variant.Light]: css`
+    background-color: ${uiColors.gray.light3};
+    border: 1px solid ${uiColors.gray.light2};
+    box-shadow: 0px 2px 4px ${transparentize(0.85, uiColors.black)};
+  `,
+};
+
+interface TooltipProps
+  extends Omit<PopoverProps, 'active' | 'spacing' | 'refEl' | 'usePortal'> {
+  /**
+   * A slot for the element used to trigger the `Tooltip`.
+   */
+  trigger: React.ReactElement | Function;
+
+  /**
+   * Determines if a `hover` or `click` event will trigger the opening of a `Tooltip`.
+   */
+  triggerEvent?: TriggerEvent;
+
+  /**
+   * Controls component and determines the open state of the `Tooltip`
+   *
+   * default: `false`
+   */
+  open?: boolean;
+
+  /**
+   * Callback to change the open state of the `Tooltip`.
+   */
+  setOpen?: (
+    open: boolean,
+  ) => void | React.Dispatch<React.SetStateAction<boolean>>;
+
+  /**
+   * Whether the `Tooltip` will be `light` or `dark`.
+   */
+  variant?: Variant;
+
+  /**
+   * id given to `Tooltip` content.
+   */
+  id?: string;
+
+  /**
+   * Callback to determine whether or not `Tooltip` should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+}
+
+/**
+ * # Tooltip
+ *
+ * Tooltip component
+ *
+ * ```
+<Tooltip
+  align='top'
+  justify='start'
+  trigger={<button>trigger</button>}
+  triggerEvent='hover'
+  variant='light'
+>
+  I am an uncontrolled Tooltip!
+</Tooltip>
+```
+ * @param props.children Content to appear inside of Tooltip.
+ * @param props.open Boolean to describe whether or not Tooltip is open.
+ * @param props.setOpen Callback to change the open state of the Tooltip.
+ * @param props.variant Whether the Tooltip should be `dark` or `light`.
+ * @param props.className Classname applied to Tooltip.
+ * @param props.align Alignment of Tooltip relative to trigger: `top`, `bottom`, `left`, `right`.
+ * @param props.justify Justification of Tooltip relative to trigger: `start`, `middle`, `end`.
+ * @param props.trigger Trigger element can be ReactNode or function.
+ * @param props.triggerEvent Whether the Tooltip should be triggered by a `click` or `hover`.
+ * @param props.id id given to Tooltip content.
+ */
+function Tooltip({
+  open: controlledOpen,
+  setOpen: controlledSetOpen,
+  className,
+  children,
+  trigger,
+  variant = Variant.Light,
+  triggerEvent = TriggerEvent.Hover,
+  align = 'top',
+  justify = 'start',
+  id,
+  shouldClose,
+  ...rest
+}: TooltipProps) {
+  const isControlled = !!controlledSetOpen;
```

Comment:
Not a fan of the double negative. Hurts my brain.

Can we do `typeof controlledSetOpen === 'function'` to say what we actually want?

Reply:
Also I think control is handled by whether `open` is defined, i.e. `isControlled = typeof open === 'boolean'`. 

If you set a setOpen function and then don't set open then it's controlled but nothing is defining what it's controlled behavior is.

If you set open and omit setOpen then the value is set but then just is never updated.

Reply:
👍